actor/v7action/service_instance_test.go

```diff
@@ -133,48 +134,70 @@ var _ = Describe("Service Instance Actions", func() {
 					ccv3.Warnings{"some-service-instance-warning"},
 					nil,
 				)
+
+				fakeCloudControllerClient.GetServiceInstanceParametersReturns(
+					types.NewOptionalObject(map[string]interface{}{"foo": "bar"}),
+					ccv3.Warnings{"some-parameters-warning"},
+					nil,
+				)
 			})
 
-			It("returns a service instance with relationships and warnings", func() {
-				Expect(executionError).NotTo(HaveOccurred())
-				Expect(warnings).To(ConsistOf("some-service-instance-warning"))
+			It("returns a service instance with details and warnings", func() {
+				By("returning warnings and no errors", func() {
+					Expect(executionError).NotTo(HaveOccurred())
+					Expect(warnings).To(ConsistOf(
+						"some-service-instance-warning",
+						"some-parameters-warning",
+					))
+				})
 
-				Expect(serviceInstance).To(Equal(
-					ServiceInstanceWithRelationships{
-						ServiceInstance: resources.ServiceInstance{
-							Type: resources.ManagedServiceInstance,
-							Name: "some-service-instance",
-							GUID: "some-service-instance-guid",
+				By("returning the correct service instance details object", func() {
+					Expect(serviceInstance).To(Equal(
+						ServiceInstanceDetails{
+							ServiceInstance: resources.ServiceInstance{
+								Type: resources.ManagedServiceInstance,
+								Name: serviceInstanceName,
+								GUID: serviceInstanceGUID,
+							},
+							ServiceOffering: resources.ServiceOffering{
+								Name:             serviceOfferingName,
+								Description:      serviceOfferingDescription,
+								DocumentationURL: serviceOfferingDocumentation,
+							},
+							ServicePlanName:         servicePlanName,
+							ServiceBrokerName:       serviceBrokerName,
+							SharedStatus:            SharedStatus{},
+							Parameters:              types.NewOptionalObject(map[string]interface{}{"foo": "bar"}),
+							ParametersMissingReason: "",
 						},
-						ServiceOffering: resources.ServiceOffering{
-							Name:             serviceOfferingName,
-							Description:      serviceOfferingDescription,
-							DocumentationURL: serviceOfferingDocumentation,
+					))
+				})
+
+				By("making the correct call to get the service instance", func() {
```

Comment:
Just an opinion regarding the use of By. I find it more readable to have separate It blocks. I see this worth it in an integration test where set up and tear down are so costly. But cli unit test run so fast that I'd rather have the separate It. 
Also, passing a function to By makes it so verbose, maybe just leaving the function out (as it is optional) would reduce the clutter?